typescript-react/src/components/CardListTitleMenu.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useEffect } from "react";
+import Modal from "react-modal";
+import ShareFolderModal from "./Modal/ShareFolderModal";
+import RenameFolderModal from "./Modal/RenameFolderModal";
+import DeleteFolderModal from "./Modal/DeleteFolderModal";
+import "../style/CardListTitleMenu.css";
+
+interface CardListTitleMenuProps {
+  name: string;
+}
+
+const CardListTitleMenu: React.FC<CardListTitleMenuProps> = ({ name }) => {
+  const isAllFolderButton = name === "전체";
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modalType, setModalType] = useState("");
+
+  const handleOpenModal = (type: string) => {
+    setModalType(type);
+    setIsModalOpen(true);
+  };
+
+  const handleCloseModal = () => {
+    setIsModalOpen(false);
+  };
+
+  useEffect(() => {
+    Modal.setAppElement("#root");
+  }, []);
+
+  return (
+    <div className="cardlisttitlemenu-section">
+      <div className="cardlisttitlemenu-title">{name}</div>
+      <div className="cardlisttitlemenu-option">
+        {isAllFolderButton ? null : (
+          <>
+            <button
+              className="cardlisttitlemenu-btn"
+              onClick={() => handleOpenModal("share")}
+            >
+              {" "}
+              <img
+                className="cardlisttitlemenu-btn-icon"
+                src="src/assets/share.svg"
+                alt="공유 버튼"
+              />
+              공유
+            </button>
+            <button
+              className="cardlisttitlemenu-btn"
+              onClick={() => handleOpenModal("rename")}
+            >
+              {" "}
+              <img
+                className="cardlisttitlemenu-btn-icon"
+                src="src/assets/pen.svg"
+                alt="이름 변경 버튼"
+              />
+              이름변경
+            </button>
+            <button
+              className="cardlisttitlemenu-btn"
+              onClick={() => handleOpenModal("delete")}
+            >
+              {" "}
+              <img
+                className="cardlisttitlemenu-btn-icon"
+                src="src/assets/Group 36.svg"
+                alt="삭제 버튼"
+              />
+              삭제
+            </button>
+          </>
+        )}
+      </div>
+
+      {isModalOpen && modalType === "share" && (
+        <Modal isOpen={isModalOpen}>
+          <ShareFolderModal name={name} onClose={handleCloseModal} />
+        </Modal>
+      )}
+      {isModalOpen && modalType === "rename" && (
+        <Modal isOpen={isModalOpen}>
+          <RenameFolderModal name={name} onClose={handleCloseModal} />
+        </Modal>
+      )}
+      {isModalOpen && modalType === "delete" && (
+        <Modal isOpen={isModalOpen}>
+          <DeleteFolderModal name={name} onClose={handleCloseModal} />
+        </Modal>
+      )}
```

Comment:
조건식을 판별하는 함수로 구분해서 반환되는 값에 따라 JSX를 렌더 하는방식으로도 수정할 수 있어요.

```suggestion
const renderModalContent = () => {
  switch (modalType) {
    case "share":
      return <ShareFolderModal name={name} onClose={handleCloseModal} />;
    case "rename":
      return <RenameFolderModal name={name} onClose={handleCloseModal} />;
    case "delete":
      return <DeleteFolderModal name={name} onClose={handleCloseModal} />;
    default:
      return null;
  }
};

return (
...생략
       삭제
            </button>
          </>
        )}
      </div>
      
    {isModalOpen && (
      <Modal isOpen={isModalOpen}>
        {renderModalContent()}
      </Modal>
    )}

);
```

이렇게 하면 JSX부분은 좀더 이해하기 쉬워지고 나중에 조건이 추가되면 switch Case문에서 case만 추가하거나, 수정하기 쉬워지기도 합니다.